jujugui/static/gui/src/app/components/inspector-relations-item/inspector-relations-item.js

```diff
@@ -38,19 +52,77 @@ YUI.add('inspector-relations-item', function() {
       });
     },
 
+    /**
+      Handles the checkbox change action by either calling the parent supplied
+      whenChanged method and by setting the local checked state.
+
+      @method _handleChange
+      @param {Object} The change event from the checkbox.
+    */
+    _handleChange: function(e) {
+      var whenChanged = this.props.whenChanged;
+      var checked = e.currentTarget.checked;
+      this.setState({checked: checked}, () => {
+        // When whenChanged is set by the list parent and is used to (de)select
+        // all checkboxes. It is called in the setState callback so that the
+        // updated state is available if we inspect it from whenChanged.
+        if (whenChanged) {
+          whenChanged(checked);
+        }
+      });
+    },
+
+    /**
+      Don't bubble the click event to the parent.
+
+      @method _stopBubble
+      @param {Object} The click event from the checkbox.
+    */
+    _stopBubble: function(e) {
+      e.stopPropagation();
+    },
+
+    /**
+      Returns the id if the item is not a navigation element.
+
+      @method _valueClasses
+      @param {String} id The id of the checkbox.
+      @returns {String} The id of the element or a blank string.
+    */
+    _generateId: function(id) {
```

Comment:
Thing is we dont want the `htmlFor` to match the input because then it will tick the checkbox as it navigates to the details view. This is the same in the units view in the inspector.